packages/core/src/resolveTrustChains/resolveTrustChains.ts

```diff
@@ -59,24 +65,92 @@ export const resolveTrustChains = async (options: Options): Promise<Array<TrustC
 
     if (entityStatementChain.some((statement) => statement.exp < now)) {
       // Skip expired chains
+      // TODO: Think about how we want to share this conclusion with the caller
       continue
     }
 
-    // TODO: Merge all the policies and check them against the metadata of the leaf entity
+    if (entityStatementChain.length === 1) {
+      // When there is only one statement, we can assume that the leaf is also the trust anchor
+      const leafEntityConfiguration = entityConfigurationChain[0]
+      if (!leafEntityConfiguration)
+        throw new OpenIdFederationError(PolicyErrorStage.Validation, 'No leaf entity configuration found')
+
+      trustChains.push({
+        chain: entityStatementChain,
+        trustAnchorEntityConfiguration: leafEntityConfiguration,
+        rawLeafEntityConfiguration: leafEntityConfiguration,
+        resolvedLeafMetadata: leafEntityConfiguration.metadata,
+      })
+      continue
+    }
 
     const leafEntityConfiguration = entityConfigurationChain[0]
-    // Should never happen but for the type safety
     if (!leafEntityConfiguration)
-      throw new OpenIdFederationError(ErrorCode.Validation, 'No leaf entity configuration found')
+      throw new OpenIdFederationError(PolicyErrorStage.Validation, 'No leaf entity configuration found')
+
+    const statementsWithoutLeaf = entityStatementChain.slice(0, -1)
+    const combinedPolicyResult = await tryCatch(async () =>
+      combineMetadataPolicies({
+        statements: statementsWithoutLeaf,
+      })
+    )
+    if (!combinedPolicyResult.success) {
+      if (combinedPolicyResult.error instanceof PolicyOperatorMergeError) {
+        // When some operators can't be merged, we can declare the chain invalid
+        // TODO: Think about how we want to share this conclusion with the caller
+        continue
+      }
+      if (OpenIdFederationError.isMetadataPolicyCritError(combinedPolicyResult.error)) {
+        // When the error is a metadata_policy_crit error, we can declare the chain invalid
+        // TODO: Think about how we want to share this conclusion with the caller
+        continue
+      }
+
+      throw new OpenIdFederationError(
+        PolicyErrorStage.PolicyMerge,
+        'Unexpected error while applying policy',
+        combinedPolicyResult.error
+      )
```

Comment:
shouldn't we continue to the next one in this case? One entity having broken/invalid metadata could cause all chain validation to fail. I think being permissive if we might still have other options is good in this case